src/serial.rs

```diff
@@ -377,11 +377,14 @@ impl<T: Trigger, EV: SerialEvents, W: Write> Serial<T, EV, W> {
                         self.received_data_interrupt().map_err(Error::Trigger)?;
                     }
                 } else {
-                    self.out.write_all(&[value]).map_err(Error::IOError)?;
-                    self.out.flush().map_err(Error::IOError)?;
-                    self.serial_evts.out_byte();
-
+                    let res = self
```

Comment:
We should also add an integration test to test this scenario in which the out is full, but the serial is not blocked (i.e. thre is sent).